### PR TITLE
Improve compatibility for array operands

### LINKED ISSUE #189: Printing an array outputs garbage and deviates from reference
https://shopify.github.io/liquid/filters/slice/

The code
```liquid
{% assign beatles = "John, Paul, George, Ringo" | split: ", " %}
{{ beatles | slice: 1, 2 }}
```

Expected output
```
PaulGeorge
```

Actual output 
```
 Array ( [0] => Paul [1] => George ) 
```

### PR DESCRIPTION
**Pull request by live627**
Fix #189

### CODE REVIEW

**Review by sanmai — CHANGES_REQUESTED** [review]
How can we avoid introducing XSS?

File: src/Liquid/AbstractBlock.php [hunk 1]
```diff
@@ -164,7 +164,7 @@ protected function renderAll(array $list, Context $context)
 			}
 
 			if (is_array($value)) {
-				$value = htmlspecialchars(print_r($value, true));
+				$value = implode($value);
```
Comment (sanmai) [inline src/Liquid/AbstractBlock.php:167]: This used to escape HTML; we have to keep it this way

**Review by sanmai — CHANGES_REQUESTED** [review]

File: src/Liquid/AbstractBlock.php (hunk 1 above)
Comment (live627) [reply src/Liquid/AbstractBlock.php]: done

**Review by sanmai — APPROVED** [review]
Nice work!

**Pull request merged**